contracts/cw721-stake/src/contract.rs

```diff
@@ -0,0 +1,527 @@
+                .map(|token_id| -> StdResult<WasmMsg> {
+                    Ok(cosmwasm_std::WasmMsg::Execute {
+                        contract_addr: config.nft_address.to_string(),
+                        msg: to_binary(&cw721::Cw721ExecuteMsg::TransferNft {
+                            recipient: info.sender.to_string(),
+                            token_id,
+                        })?,
+                        funds: vec![],
+                    })
+                })
+                .collect::<StdResult<Vec<_>>>()?;
+
+            Ok(Response::default()
+                .add_messages(return_messages)
+                .add_submessages(hook_msgs)
+                .add_attribute("action", "unstake")
+                .add_attribute("from", info.sender)
+                .add_attribute("claim_duration", "None"))
+        }
+
+        Some(duration) => {
+            let outstanding_claims = NFT_CLAIMS
+                .query_claims(deps.as_ref(), &info.sender)?
+                .nft_claims;
+            if outstanding_claims.len() >= MAX_CLAIMS as usize {
+                return Err(ContractError::TooManyClaims {});
+            }
+
+            // Out of gas here is fine - just try again with fewer
+            // tokens.
+            NFT_CLAIMS.create_nft_claims(
+                deps.storage,
+                &info.sender,
+                token_ids,
+                duration.after(&env.block),
+            )?;
+
+            Ok(Response::default()
+                .add_attribute("action", "unstake")
+                .add_submessages(hook_msgs)
+                .add_attribute("from", info.sender)
+                .add_attribute("claim_duration", format!("{}", duration)))
+        }
+    }
+}
+
+pub fn execute_claim_nfts(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let nfts = NFT_CLAIMS.claim_nfts(deps.storage, &info.sender, &_env.block)?;
+    if nfts.is_empty() {
+        return Err(ContractError::NothingToClaim {});
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+
+    let msgs = nfts
+        .into_iter()
+        .map(|nft| -> StdResult<CosmosMsg> {
+            Ok(WasmMsg::Execute {
+                contract_addr: config.nft_address.to_string(),
+                msg: to_binary(&cw721::Cw721ExecuteMsg::TransferNft {
+                    recipient: info.sender.to_string(),
+                    token_id: nft,
+                })?,
+                funds: vec![],
+            }
+            .into())
+        })
+        .collect::<StdResult<Vec<_>>>()?;
+
+    Ok(Response::default()
+        .add_messages(msgs)
+        .add_attribute("action", "claim_nfts")
+        .add_attribute("from", info.sender))
+}
+
+pub fn execute_update_config(
+    info: MessageInfo,
+    deps: DepsMut,
+    new_owner: Option<String>,
+    new_manager: Option<String>,
+    duration: Option<Duration>,
+) -> Result<Response, ContractError> {
+    let new_owner = new_owner
```

Comment:
I know the other staking contract follows this owner / manager pattern, but since AFAIK none of our other modules do that, what's the reasoning to add it to the staking contracts in particular?

Reply:
Good question. The original intent was to allow a DAO to delegate the ability to change the unstaking duration to a subDAO. At this point I think it's just become a pattern I do and don't think terribly critically about. What do you think? It is worth keeping around?

Reply:
I understand that use case, but I think I would err on the side of removing it for now since it seems a bit like we are over-engineering for future scenarios. My main concern was just that the admin story seems to be getting more muddled over time with several different types of admins (the wasm admin, the subdao admin, owners, managers) -- I think we could take a step back and think about the different types of roles / authorizations that we want to support, taking into consideration the authorization middleware as well.

Reply:
> I understand that use case, but I think I would err on the side of removing it for now since it seems a bit like we are over-engineering for future scenarios.

Can do this as a separate issue / ticket. We should make whatever we decide consistent everywhere.